### PR TITLE
better organize quarkus-extensions variables in build-quarkus-app.sh

### PR DESCRIPTION
**Pull request by rgdoliveira**
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [ ] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: scripts/logic/build-quarkus-app.sh [hunk 1]
```diff
@@ -17,11 +17,9 @@ quarkus_platform_version="${3}"
...
+quarkus_extensions="quarkus-kubernetes,kogito-quarkus-serverless-workflow,kogito-addons-quarkus-knative-eventing,smallrye-health,org.kie.kogito:kogito-addons-quarkus-fabric8-kubernetes-service-catalog:${kogito_version},org.kie.kogito:kogito-addons-quarkus-kubernetes:${kogito_version}"
 # dev mode purpose extensions used only by the kogito-swf-devmode
-kogito_swf_devmode_extensions='kogito-quarkus-serverless-workflow-devui,kogito-addons-quarkus-source-files'
-# devmode extensions that are not available in RHBQ 2.13.x
-kogito_swf_devmode_extensions_2_16="org.kie.kogito:kogito-addons-quarkus-jobs-service-embedded:${kogito_version},org.kie.kogito:kogito-addons-quarkus-data-index-inmemory:${kogito_version},org.kie.kogito:kogito-addons-quarkus-fabric8-kubernetes-service-catalog:${kogito_version},org.kie.kogito:kogito-addons-quarkus-kubernetes:${kogito_version}"
+kogito_swf_devmode_extensions="kogito-quarkus-serverless-workflow-devui,kogito-addons-quarkus-source-files,org.kie.kogito:kogito-addons-quarkus-jobs-service-embedded:${kogito_version},org.kie.kogito:kogito-addons-quarkus-data-index-inmemory:${kogito_version}"
```
Comment (rgdoliveira) [inline scripts/logic/build-quarkus-app.sh:22]: PR https://github.com/kiegroup/kogito-images/pull/1575 added 2 new extensions `kogito-addons-quarkus-fabric8-kubernetes-service-catalog` and `kogito-addons-quarkus-kubernetes`.

Could you please confirm if those extensions are required for both swf-devmode and swf-builder images artifacts?

**Review by davidesalerno — APPROVED** [review]

File: scripts/logic/build-quarkus-app.sh (hunk 1 above)
Comment (ricardozanini) [reply scripts/logic/build-quarkus-app.sh:22]: Thank u for this PR! Yes, they are.

**Review by ricardozanini — APPROVED** [review]

**Comment by kie-ci1** [comment]
**PR job** `#903` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/display/redirect

**Test results:**
- PASSED: 257
- FAILED: 13

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_jobs_service_postgresql___Build/Test_kogito_jobs_service_postgresql___Verify_if_the_properties_were_correctly_set_when_aren't_passed/">Build & Test Images / Build&Test kogito-jobs-service-postgresql / Build/Test kogito-jobs-service-postgresql / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the properties were correctly set when aren't passed</a></summary>
HOOK-ERROR in after_scenario: ReadTimeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_jobs_service_postgresql___Build/Test_kogito_jobs_service_postgresql___Verify_if_Java_Remote_Debug_is_correctly_configured/">Build & Test Images / Build&Test kogito-jobs-service-postgresql / Build/Test kogito-jobs-service-postgresql / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if Java Remote Debug is correctly configured</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_data_index_infinispan___Build/Test_kogito_data_index_infinispan___Verify_if_Java_Remote_Debug_is_correctly_configured/">Build & Test Images / Build&Test kogito-data-index-infinispan / Build/Test kogito-data-index-infinispan / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if Java Remote Debug is correctly configured</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_data_index_infinispan___Build/Test_kogito_data_index_infinispan___Verify_if_the_DEFAULT_MEM_RATIO_properties_are_overridden_with_different_values_from_user_provided_Xmx_and_Xms/">Build & Test Images / Build&Test kogito-data-index-infinispan / Build/Test kogito-data-index-infinispan / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the DEFAULT MEM RATIO properties are overridden with different values from user provided Xmx and Xms</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_trusty_postgresql___Build/Test_kogito_trusty_postgresql___Verify_if_the_DEFAULT_MEM_RATIO_properties_are_overridden_with_different_values/">Build & Test Images / Build&Test kogito-trusty-postgresql / Build/Test kogito-trusty-postgresql / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the DEFAULT MEM RATIO properties are overridden with different values</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_trusty_postgresql___Build/Test_kogito_trusty_postgresql___Verify_if_the_properties_were_correctly_set_when_aren't_passed/">Build & Test Images / Build&Test kogito-trusty-postgresql / Build/Test kogito-trusty-postgresql / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the properties were correctly set when aren't passed</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_data_index_mongodb___Build/Test_kogito_data_index_mongodb___Verify_if_Java_Remote_Debug_is_correctly_configured/">Build & Test Images / Build&Test kogito-data-index-mongodb / Build/Test kogito-data-index-mongodb / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if Java Remote Debug is correctly configured</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_data_index_mongodb___Build/Test_kogito_data_index_mongodb___Verify_if_the_DEFAULT_MEM_RATIO_properties_are_overridden_with_different_values_from_user_provided_Xmx_and_Xms/">Build & Test Images / Build&Test kogito-data-index-mongodb / Build/Test kogito-data-index-mongodb / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the DEFAULT MEM RATIO properties are overridden with different values from user provided Xmx and Xms</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_trusty_infinispan___Build/Test_kogito_trusty_infinispan___Verify_if_the_DEFAULT_MEM_RATIO_properties_are_overridden_with_different_values/">Build & Test Images / Build&Test kogito-trusty-infinispan / Build/Test kogito-trusty-infinispan / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the DEFAULT MEM RATIO properties are overridden with different values</a></summary>
HOOK-ERROR in after_scenario: ReadTimeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_trusty_infinispan___Build/Test_kogito_trusty_infinispan___Verify_if_the_properties_were_correctly_set_when_aren't_passed/">Build & Test Images / Build&Test kogito-trusty-infinispan / Build/Test kogito-trusty-infinispan / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the properties were correctly set when aren't passed</a></summary>
UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_data_index_postgresql___Build/Test_kogito_data_index_postgresql___Verify_if_the_properties_were_correctly_set_when_aren't_passed/">Build & Test Images / Build&Test kogito-data-index-postgresql / Build/Test kogito-data-index-postgresql / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the properties were correctly set when aren't passed</a></summary>
HOOK-ERROR in after_scenario: TypeError: a bytes-like object is required, not 'str'
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.common-dynamic-resources/Common tests for Kogito images/Build___Test_Images___Build_Test_kogito_jobs_service_infinispan___Build/Test_kogito_jobs_service_infinispan___Verify_if_the_properties_were_correctly_set_when_aren't_passed/">Build & Test Images / Build&Test kogito-jobs-service-infinispan / Build/Test kogito-jobs-service-infinispan / features.image.common-dynamic-resources.Common tests for Kogito images.Verify if the properties were correctly set when aren't passed</a></summary>
HOOK-ERROR in after_scenario: TypeError: a bytes-like object is required, not 'str'
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/903/testReport/features.image.kogito-common-builder-jvm/kogito-s2i-builder image JVM build tests/Perform_a_second_incremental_s2i_build_using_quarkus_runtime_type/">features.image.kogito-common-builder-jvm.kogito-s2i-builder image JVM build tests.Perform a second incremental s2i build using quarkus runtime type</a></summary>
Phrase 'Expanding artifacts from incremental build...' was not found in the output of S2I
</details>

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
failures unrelated

**Comment by rgdoliveira** [comment]
Backport PR https://github.com/kiegroup/kogito-images/pull/1586 merged in `1.40.x` branch

**Pull request merged**